src/Tasks/FileAccessMutex.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.Build.Tasks
+{
+    internal sealed class FileAccessMutex : IDisposable
```

Comment:
This doesn't seem to be specific to file access, it is a general purpose system-wide mutex primitive that can be used in a variety of scenarios. 

This design differs from the sample I gave and I'm not sure the IsLocked is thread safe. What is the rationale for the current design?